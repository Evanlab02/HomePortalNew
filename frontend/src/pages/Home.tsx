import { useEffect, useState } from "react";
import { FlexGridItem, FlexGrid, Card } from "@evanlab02/hp-react-components";
import { getApps } from "../api/ApplicationController";
import { HomePortalApplication } from "../interfaces/ApplicationInterfaces";
import "./styles/Home.scss";

export default function Home() {
	const [apps, setApps] = useState<HomePortalApplication[]>([]);

	useEffect(() => {
		void fetchApplications();
	}, []);

	const fetchApplications = async () => {
		const applications = await getApps();
		setApps(applications);
	};

	return (
		<div className="home-portal" data-testid="hp-home-page">
			<h1>Services</h1>
			<FlexGrid>
				{apps.map((app) => {
					return (
						<FlexGridItem xs={12} sm={12} md={12} lg={6} xl={4} xxl={2} key={`${app.title}-${app.link_name}`}>
							<Card
								title={app.title}
								extra={<a target="_blank" className="hp-service-link" href={app.link} rel="noreferrer">{app.link_name}</a>}
							>
								<p>{app.description}</p>
							</Card>
						</FlexGridItem>
					);
				})}
			</FlexGrid>
		</div>
	);
}