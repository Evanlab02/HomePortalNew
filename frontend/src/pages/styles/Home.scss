.home-portal {
    margin: 1rem;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial, 'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';

    .ant-card {
        margin: 1rem;

        .hp-service-link {
            color: #69b1ff;
        }
    }
}