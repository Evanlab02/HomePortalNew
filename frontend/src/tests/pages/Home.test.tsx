import { render } from "@testing-library/react";
import { beforeEach, expect, test } from "vitest";
import Home from "../../pages/Home";
import { appData } from "../mock/ApplicationData.mock";

beforeEach(() => {
	document.getElementsByTagName("html")[0].innerHTML = "";
});

test("Home page renders correctly", async () => {
	fetchMock.mockResponseOnce(JSON.stringify(appData));

	const { findByTestId } = render(
		<Home />
	);

	const home = await findByTestId("hp-home-page");
	expect(home).toMatchSnapshot();
});