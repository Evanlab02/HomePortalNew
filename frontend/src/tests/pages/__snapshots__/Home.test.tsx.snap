// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page renders correctly 1`] = `
<div
  class="home-portal"
  data-testid="hp-home-page"
>
  <h1>
    Services
  </h1>
  <div
    class="hp-flex-grid-container"
    data-testid="hp-flex-grid-container"
  >
    <div
      class="hp-flex-grid-col-xs-12 hp-flex-grid-col-sm-12 hp-flex-grid-col-md-12 hp-flex-grid-col-lg-6 hp-flex-grid-col-xl-4 hp-flex-grid-col-xxl-2"
      data-testid="hp-flex-grid-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-w5lxuz"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Postgres Administration
            </div>
            <div
              class="ant-card-extra"
            >
              <a
                class="hp-service-link"
                href="/pgadmin/"
              >
                PgAdmin
              </a>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <p>
            Intended for developers. Manage the postgres database provided by home portal and any other database that might be running.
          </p>
        </div>
      </div>
    </div>
    <div
      class="hp-flex-grid-col-xs-12 hp-flex-grid-col-sm-12 hp-flex-grid-col-md-12 hp-flex-grid-col-lg-6 hp-flex-grid-col-xl-4 hp-flex-grid-col-xxl-2"
      data-testid="hp-flex-grid-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-w5lxuz"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Home Portal Administration
            </div>
            <div
              class="ant-card-extra"
            >
              <a
                class="hp-service-link"
                href="/admin/"
              >
                Admin
              </a>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <p>
            Intended for developers and home portal administrators. Manage home portal users and configuration through the built in admin interface.
          </p>
        </div>
      </div>
    </div>
    <div
      class="hp-flex-grid-col-xs-12 hp-flex-grid-col-sm-12 hp-flex-grid-col-md-12 hp-flex-grid-col-lg-6 hp-flex-grid-col-xl-4 hp-flex-grid-col-xxl-2"
      data-testid="hp-flex-grid-item"
    >
      <div
        class="ant-card ant-card-bordered css-dev-only-do-not-override-w5lxuz"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              Accounts
            </div>
            <div
              class="ant-card-extra"
            >
              <a
                class="hp-service-link"
                href="/accounts/"
              >
                My Account
              </a>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <p>
            Manage your account on home portal.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
